syntax = "proto3";
package intelligent;

message Lamp {
    string id = 1;
    bool status = 2;
    string type = 3;
    string color = 4;
    repeated string cmds = 5;
}

message Treadmill {
    string id = 1;
    bool status = 2;
    string type = 3;
    double dist = 4;
    float vel = 5;
    repeated string cmds = 6;
}

message AC {
    string id = 1;
    bool status = 2;
    string type = 3;
    int32 temp = 4;
    repeated string cmds = 5;
}

message Response {
    repeated Lamp lamps = 1;
    repeated Treadmill treadmills = 2;
    repeated AC acs = 3;
}

message Request {
    string cmd = 1;
    string id_obj = 2;
    repeated string args = 3; 
    string type = 4;
}



message ResponseSingleObject {

    string id_obj = 1;
    oneof obj {
        Lamp objL = 2;
        Treadmill objT = 3;
        AC objA = 4;
    }
        
}

message ResponseLamp {

    string id_obj = 1;
    Lamp obj = 2;
}   


message ResponseTreadmill {

    string id_obj = 1;
    Treadmill obj = 2;
}

message ResponseAC {

    string id_obj = 1;
    AC obj = 2;
}